docs/src/main/asciidoc/kafka.adoc

```diff
@@ -3,438 +3,946 @@ This guide is maintained in the main Quarkus repository
+@Path("/prices")
+public class PriceResource {
 
-Create the `src/main/resources/META-INF/resources/prices.html` file, with the following content:
+    @Inject @Channel("price-create") MutinyEmitter<Double> priceEmitter;
 
-[source, html]
+    @POST
+    @Consumes(MediaType.TEXT_PLAIN)
+    public Uni<String> addPrice(Double price) {
+        return quoteRequestEmitter.send(price)
+                .map(x -> "ok")
+                .onFailure().recoverWithItem("ko");
+    }
+}
 ----
-<!DOCTYPE html>
-<html lang="en">
-<head>
-    <meta charset="UTF-8">
-    <title>Prices</title>
 
-    <link rel="stylesheet" type="text/css"
-          href="https://cdnjs.cloudflare.com/ajax/libs/patternfly/3.24.0/css/patternfly.min.css">
-    <link rel="stylesheet" type="text/css"
-          href="https://cdnjs.cloudflare.com/ajax/libs/patternfly/3.24.0/css/patternfly-additions.min.css">
-</head>
-<body>
-<div class="container">
+There’s also the ability to block on sending the event to the emitter with `sendAndAwait` method. It will only return from the method when the event is acked or nacked by the receiver.
+
+=== Write Acknowledgement
+
+When Kafka broker receives a record, its acknowledgement can take time depending on the configuration.
+Also, it stores in-memory the records that cannot be written.
 
-    <h2>Last price</h2>
-    <div class="row">
-    <p class="col-md-12">The last price is <strong><span id="content">N/A</span>&nbsp;&euro;</strong>.</p>
-    </div>
-</div>
-</body>
-<script src="https://code.jquery.com/jquery-3.3.1.min.js"></script>
-<script>
-    var source = new EventSource("/prices/stream");
-    source.onmessage = function (event) {
-        document.getElementById("content").innerHTML = event.data;
-    };
-</script>
-</html>
+By default, the connector does wait for Kafka to acknowledge the record to continue the processing (acknowledging the received Message).
+You can disable this by setting the `waitForWriteCompletion` attribute to `false`.
+
+Note that the `acks` attribute has a huge impact on the record acknowledgement.
+
+If a record cannot be written, the message is nacked.
+
+=== Backpressure
+
+The Kafka outbound connector handles back-pressure, monitoring the number of in-flight messages waiting to be written to the Kafka broker.
+The number of in-flight messages is configured using the `max-inflight-messages` attribute and defaults to 1024.
+
+The connector only sends that amount of messages concurrently.
+No other messages will be sent until at least one in-flight message gets acknowledged by the broker.
+Then, the connector writes a new message to Kafka when one of the broker’s in-flight messages get acknowledged.
+Be sure to configure Kafka’s `batch.size` and `linger.ms` accordingly.
+
+You can also remove the limit of in-flight messages by setting `max-inflight-messages` to `0`.
+However, note that the Kafka Producer may block if the number of requests reaches `max.in.flight.requests.per.connection`.
+
+=== Retries
```

Comment:
Should it be under the producing message section?

Reply:
I think we can rename this section to "Retrying message dispatch" and keep it here. I was thinking of adding the "Retrying processing" section near receiving messages/error handling. What do you think ?